extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/CommandListenerBuildItem.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.mongodb.deployment;
+
+import java.util.List;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class CommandListenerBuildItem extends SimpleBuildItem {
```

Comment:
I see this is done to be symmetrical. But I fail to see why we need build items for that, rather than just having private methods.

Is there any point for the others? In this case it makes sense to be consistent and have it here too.

Reply:
I'm not an expert but my guess would be that in case we use private methods, 
we still would need to pass the CombinedIndexBuildItem to the buildsteps, the index lists process would be run two times and also we would be postponing the class name filtering to the Static init phase instead of doing it during the augmentation.

Also having BuildItems and BuildSteps makes the flow more clear about what it is needed to build the final objects.